### DISCUSSION TITLE
[Help] Custom Product Type creation

### DISCUSSION THREAD

**Discussion by igoraugustynski** [swellstores/community#267] [Help] [answered]
Hello,
I am trying to create a new Product Type besides the defaults of Digital, Physical, Bundle and Gift Card. I know it's possible because the [documentation](https://developers.swell.is/guides/core-concepts/product-modeling#custom-product-types) mentions it, but I cannot find a way to do this. What am I missing? 
Also -> After adding some custom Content Fields to the product model, can I somehow hide/show them based on the product type selected? I want the store admins to be able to fill out different fields based on which type of product they are creating.

**Comment by logeshswell** [comment, marked as answer]
@igor-augustynski-furba You can view the available product types using the below query:
https://store-id.swell.store/admin/console?get/settings/-products~Htypes 

You can add/delete the product type as follows:
<img width="879" alt="Screenshot 2023-06-12 at 12 58 45 PM" src="https://github.com/swellstores/community/assets/100766131/c6e90244-01db-4f4a-ab56-b84c8d3e0b1c">

We will make sure to update our documentation and provide a user interface to edit these details in the admin dashboard.

**Reply by igoraugustynski** [reply]
Thanks. Can I also add custom content fields to a product type?

**Reply by logeshswell** [reply]
> Can I somehow hide/show them based on the product type selected?

No, this feature is currently not available, but I will take it as a feature request to show content fields based on the conditions like product type or category, etc.

**Reply by igoraugustynski** [reply]
So what is the current purpose of Custom Product Types?

**Reply by logeshswell** [reply]
@igor-augustynski-furba As we mentioned in the [documentation](https://developers.swell.is/guides/core-concepts/product-modeling#custom-product-types), 

> Custom product types can group together products with similar characteristics. You can define the custom type's attributes, categories, options, variant settings, and more—allowing you to streamline the process for creating new products and ensuring consistency across product families within your catalog.

**Reply by igoraugustynski** [reply]
@logeshswell I created a custom product type but I'm not seeing it when creating a new product. Shouldn't it be available here?
<img width="1244" alt="Screenshot 2023-06-13 at 19 56 47" src="https://github.com/swellstores/community/assets/94485687/6a02d23a-3d31-49dc-8a93-dd0a036b99a1">

**Reply by logeshswell** [reply]
@igor-augustynski-furba Yes, it won't be available in the admin dashboard at the moment, but it's in our pipeline.
Currently, this feature is supported via APIs.
